pytential/symbolic/stokes.py

```diff
@@ -587,77 +781,78 @@ class HsiaoKressExteriorStokesOperator(StokesOperator):
     .. automethod:: __init__
     """
 
-    def __init__(self, *, omega, alpha=None, eta=None):
+    def __init__(self, *, omega, alpha=1.0, eta=1.0, method="biharmonic",
+            mu_sym=_MU_SYM_DEFAULT, nu_sym=0.5):
         r"""
         :arg omega: farfield behaviour of the velocity field, as defined
             by :math:`A` in [HsiaoKress1985]_ Equation 2.3.
         :arg alpha: real parameter :math:`\alpha > 0`.
         :arg eta: real parameter :math:`\eta > 0`. Choosing this parameter well
             can have a non-trivial effect on the conditioning.
         """
-        super().__init__(ambient_dim=2, side=+1)
+        super().__init__(ambient_dim=2, side=+1, method=method,
+                mu_sym=mu_sym, nu_sym=nu_sym)
 
         # NOTE: in [hsiao-kress], there is an analysis on a circle, which
         # recommends values in
         #   1/2 <= alpha <= 2 and max(1/alpha, 1) <= eta <= min(2, 2/alpha)
         # so we choose alpha = eta = 1, which seems to be in line with some
         # of the presented numerical results too.
 
-        if alpha is None:
-            alpha = 1.0
-
-        if eta is None:
-            eta = 1.0
-
         self.omega = omega
         self.alpha = alpha
         self.eta = eta
 
-    def _farfield(self, mu, qbx_forced_limit):
-        length = sym.integral(self.ambient_dim, self.dim, 1)
-        return self.stokeslet.apply(
-                -self.omega / length,
-                mu,
-                qbx_forced_limit=qbx_forced_limit)
-
-    def _operator(self, sigma, normal, mu, qbx_forced_limit):
+    def _farfield(self, qbx_forced_limit):
+        source_dofdesc = sym.DOFDescriptor(None, discr_stage=sym.QBX_SOURCE_STAGE1)
+        length = sym.integral(self.ambient_dim, self.dim, 1, dofdesc=source_dofdesc)
+        # pylint does not like __new__ returning new object
+        # pylint: disable=no-member
+        result = self.stresslet.apply_stokeslet_and_stresslet(
+                -self.omega / length, [0]*self.ambient_dim, [0]*self.ambient_dim,
+                qbx_forced_limit=qbx_forced_limit, stokeslet_weight=1,
+                stresslet_weight=0)
+        # pylint: enable=no-member
+        return result
+
+    def _operator(self, sigma, normal, qbx_forced_limit):
         slp_qbx_forced_limit = qbx_forced_limit
         if slp_qbx_forced_limit == "avg":
-            slp_qbx_forced_limit = +1
+            slp_qbx_forced_limit = "avg"
```

Comment:
This doesn't do anything anymore?

Reply:
Done in e5bdc368408229d21688b6d07601d034bee3ed85